### PR TITLE
tests: lib: docsis: add stub for cm_cfg and making changes

### PR DESCRIPTION
**Pull request by mattsm**
Signed-off-by: Matthew McClintock <msm-oss@mcclintock.net>

### CODE REVIEW

File: tests/lib/docsis.py [hunk 1]
```diff
@@ -84,3 +96,68 @@ def encode_cm():
...
+    legacy_search_path = None
+
+    def __init__(self, start=None):
+        '''Creates a default basic CM cfg file for modification'''
+
+        # TODO: we require loading a file for the moment
+        if start == None:
+            raise Exception("We don't know how to start from scratch yet")
+        else:
+            # TODO: filename should not matter
```
Comment (mgualco-contractor) [inline tests/lib/docsis.py]: Nit: maybe check that the file is a text file?

File: tests/lib/docsis.py [hunk 2]
```diff
@@ -84,3 +96,68 @@ def encode_cm():
...
+            self.original_file = start
+            self.original_fname = os.path.split(start)[1]
```
Comment (mgualco-contractor) [inline tests/lib/docsis.py]: Nit: could this be `os.path.basename` ?

File: tests/lib/docsis.py (hunk 1 above)
Comment (mattsm) [reply tests/lib/docsis.py]: If it does not exist it will throw an exception with useful info...

File: tests/lib/docsis.py (hunk 2 above)
Comment (mattsm) [reply tests/lib/docsis.py]: Yes, it should... copying from input args... but for now I don't have a good test so I don't want to risk it in this commit.

**Pull request merged**